### DISCUSSION TITLE
Traceback stack does not show exact place of error (again)

### DISCUSSION THREAD

**Discussion by moser** [fastapi/fastapi#9711] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import uvicorn
from fastapi import FastAPI, Depends


app = FastAPI()


def get_something_sync():
    raise ValueError
    yield True


async def get_something_async():
    raise ValueError
    yield True


@app.get("/1")
def router_func(dependency=Depends(get_something_sync)):
    """
    BAD stacktrace
    ....
    File "/usr/lib/python3.11/contextlib.py", line 638, in enter_async_context
      result = await _enter(cm)
               ^^^^^^^^^^^^^^^^
    File "/usr/lib/python3.11/contextlib.py", line 204, in __aenter__
      return await anext(self.gen)
             ^^^^^^^^^^^^^^^^^^^^^
    File ".../.venv/lib/python3.11/site-packages/fastapi/concurrency.py", line 36, in contextmanager_in_threadpool
      raise e
    """
    return


@app.get("/2")
def router_func(dependency=Depends(get_something_async)):
    """
    GOOD stacktrace
    ...
    File "/usr/lib/python3.11/contextlib.py", line 638, in enter_async_context
      result = await _enter(cm)
               ^^^^^^^^^^^^^^^^
    File "/usr/lib/python3.11/contextlib.py", line 204, in __aenter__
      return await anext(self.gen)
             ^^^^^^^^^^^^^^^^^^^^^
    File ".../main.py", line 14, in get_something_async
      raise ValueError
    """
    return


@app.get("/3")
async def router_func(dependency=Depends(get_something_sync)):
    # BAD stacktrace
    return


@app.get("/4")
async def router_func(dependency=Depends(get_something_async)):
    # GOOD stacktrace
    return


if __name__ == "__main__":
    uvicorn.run("main:app", host="0.0.0.0", port=5600, workers=1)
```

### Description

* `python main.py`
* `curl localhost:5600/1` (and `/2`...)
* When calling the routes `/1` or `/3` I get a stack trace that does not point to the code that raised the exception (line 9). 
* When calling `/2` or `/4`, the stack trace looks ok

This is somewhat duplicated with #8428, but
* the discussion there is closed and I did not find a way to re-open it :-)
* the problem described there was a bit different (exception raised in the handler not the dependencies)
* the discussion mentions a fix (upgrade to Python 3.11.2) which seems not to work for this problem (occurs on Python 3.11.4).

### Operating System

Linux

### Operating System Details

Linux ... 5.19.0-45-generic #46~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Wed Jun 7 15:06:04 UTC 20 x86_64 

### FastAPI Version

0.97.0

### Python Version

Python 3.11.4

**Comment by arjwilliams** [comment]
This issue should be fixed by #9505.

**Comment by Kludex** [comment, marked as answer]
This is fixed on 0.101.1. 🙏
